src/systems/systems/src/electrical/emergency_generator.rs

```diff
@@ -0,0 +1,131 @@
+use super::{
+    ElectricalStateWriter, Potential, PotentialSource, ProvideFrequency, ProvidePotential,
+};
+use crate::simulation::{SimulationElement, SimulatorWriter};
+use uom::si::{electric_potential::volt, f64::*, frequency::hertz};
+
+pub struct EmergencyGenerator {
+    writer: ElectricalStateWriter,
+    running: bool,
+    is_blue_pressurised: bool,
+}
+impl EmergencyGenerator {
+    pub fn new() -> EmergencyGenerator {
+        EmergencyGenerator {
+            writer: ElectricalStateWriter::new("EMER_GEN"),
+            running: false,
+            is_blue_pressurised: false,
+        }
+    }
+
+    pub fn update(&mut self, is_blue_pressurised: bool) {
+        // TODO: The emergency generator is driven by the blue hydraulic circuit. Still to be implemented.
+        self.is_blue_pressurised = is_blue_pressurised;
```

Comment:
Still wondering what kind of hydraulic interaction we would have. First guess would be you'd get a generator RPM here from hydraulics.  Then generated potential and frequency can probably be maped from those rpm.

Reply:
Likely the emergency generator receives hydraulic pressure and can then calculate its electrical output based on that. @komp1821 will likely have the details.